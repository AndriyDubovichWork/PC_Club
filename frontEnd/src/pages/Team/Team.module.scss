.Team {
  width: 80%;
  margin: 0 auto;
  .TeamContainer {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(350px, 1fr));

    justify-items: center;

    grid-column-gap: 0px;
    grid-row-gap: 10vh;
  }
}
