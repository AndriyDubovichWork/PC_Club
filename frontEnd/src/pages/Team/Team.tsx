import React from 'react';
import SubHeader from '../../Shared/SubHeader/SubHeader';
import style from './Team.module.scss';
import Banner from '../../Shared/Banner/Banner';
import TeamMemeber from './TeamMemeber/TeamMemeber';

const Team = () => {
  const team = [
    {
      img: require('./assets/memberOne.png'),
      fullName: 'Luke Miller',
      position: 'Salesman',
    },
    {
      img: require('./assets/memberTwo.png'),
      fullName: 'Michael Diaz',
      position: 'PC Expert',
    },
    {
      img: require('./assets/memberThree.png'),
      fullName: 'Briana Ross',
      position: 'Photographer',
    },
  ];
  return (
    <>
      <main className={style.Team}>
        <SubHeader title='Team' />
        <div className={style.TeamContainer}>
          {team.map((member) => (
            <TeamMemeber {...member} key={member.fullName} />
          ))}
        </div>
      </main>
      <Banner />
    </>
  );
};

export default Team;
