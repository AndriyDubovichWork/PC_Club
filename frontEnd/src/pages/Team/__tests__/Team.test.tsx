import { render, screen, cleanup } from '@testing-library/react';
import Team from '../Team';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('Team match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/team']}>
        <Routes>
          <Route path='/PC_Club/team' element={<Team />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Team contains main tag', () => {
  const { getByRole } = render(
    <MemoryRouter initialEntries={['/PC_Club/team']}>
      <Routes>
        <Route path='/PC_Club/team' element={<Team />} />
      </Routes>
    </MemoryRouter>
  );

  expect(getByRole('main')).toBeInTheDocument();
});
