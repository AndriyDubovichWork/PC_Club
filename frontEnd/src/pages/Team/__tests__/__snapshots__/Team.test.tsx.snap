// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team match snapshot 1`] = `
Array [
  <main
    className="Team"
  >
    <img
      className="backGroundPC"
      loading="lazy"
    />
    <div
      className="Text"
    >
      <h1
        className="Title"
      >
        Team
      </h1>
      <nav
        aria-label="breadcrumb"
        className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            className="MuiBreadcrumbs-li"
          >
            <a
              aria-current="page"
              className="Link active"
              href="/PC_Club"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            aria-hidden={true}
            className="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            className="MuiBreadcrumbs-li"
          >
            <a
              aria-current="page"
              className="Link active"
              href="/PC_Club/team"
              onClick={[Function]}
            >
              Team
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      className="TeamContainer"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 teamMemeber css-1ps6pg7-MuiPaper-root"
      >
        <img
          className="image"
          loading="lazy"
          src="memberOne.png"
        />
        <div
          className="text"
        >
          <h1
            className="fullName"
          >
            Luke Miller
          </h1>
          <h3
            className="position"
          >
            Salesman
          </h3>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 teamMemeber css-1ps6pg7-MuiPaper-root"
      >
        <img
          className="image"
          loading="lazy"
          src="memberTwo.png"
        />
        <div
          className="text"
        >
          <h1
            className="fullName"
          >
            Michael Diaz
          </h1>
          <h3
            className="position"
          >
            PC Expert
          </h3>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 teamMemeber css-1ps6pg7-MuiPaper-root"
      >
        <img
          className="image"
          loading="lazy"
          src="memberThree.png"
        />
        <div
          className="text"
        >
          <h1
            className="fullName"
          >
            Briana Ross
          </h1>
          <h3
            className="position"
          >
            Photographer
          </h3>
        </div>
      </div>
    </div>
  </main>,
  <div
    className="Banner"
  >
    <h1
      className="Title"
    >
      Save big with our cheap PC Club
    </h1>
    <h3
      className="subTitle"
    >
      Top PCs. Global Suppliers.
      <span
        className="HighLighted"
      >
         24/7
      </span>
       Support.
    </h3>
  </div>,
]
`;
